spec/features/moderation/legislation_proposals_spec.rb

```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+describe "Moderate legislation proposals" do
+
+  scenario "Hide", :js do
+    citizen               = create(:user)
+    legislation_process   = create(:legislation_process)
+    legislation_proposal  = create(:legislation_proposal, legislation_process_id: legislation_process.id)
+    moderator             = create(:moderator)
+
+    login_as(moderator.user)
+    visit legislation_process_proposal_path(legislation_process, legislation_proposal)
+
+    within("#legislation_proposal_#{legislation_proposal.id}") do
+      accept_confirm { click_link "Hide" }
+    end
+
+    expect(page).to have_css("#legislation_proposal_#{legislation_proposal.id}")
```

Comment:
I think in the other tests in the moderation section there's an extra `.faded` class here :thinking:. Now when I click the link, there's no visual feedback :thinking:.

Although it isn't present in similar tests, another expectation like `expect(page).not_to have_link "Hide"` would be great :smile:.

We'd probably need to add a view like `app/views/moderation/legislation/proposals/hide.js.erb`, similar to the other views in the moderation section :thinking:, and then add some JavaScript code to `app/assets/javascripts/moderator_legislation_proposals.js`. Make sure it is **not** a CoffeeScript file :pray:. For a reference JavaScript file, have a look at: https://github.com/consul/consul/blob/master/app/assets/javascripts/moderator_debates.js

What do you think?

Reply:
Hello @javierm , I did the modifications accordingly. And added the javascripts part.